scqubits/tests/diag_tests.py

```diff
@@ -0,0 +1,281 @@
+import numpy as np
+import importlib
+import warnings
+import pytest
+import scqubits as scq
+import scipy as sp
+from scqubits import Fluxonium, Transmon, HilbertSpace
+
+diag_methods = scq.DIAG_METHODS.keys()
+
+def _library_installed(library):
+    try:
+        importlib.import_module(library)
+        return True
+    except ImportError:  # `ModuleNotFoundError` is a subclass of `ImportError`. Here, the general class of errors are checked.
+        return False
+
+def _get_library_methods(library, evals=True):
+    library_diag_methods = [
+        method for method in diag_methods
+        if method.split('_')[1] == library
+        and method.split('_')[0] == ('evals' if evals else 'esys')
+        and '_shift-inverse' not in method
+        and '_SM' not in method  # TODO: this is temporary until we find a fix for adding EJ to the Hamiltonian.
```

Comment:
For now, I have also omitted `_SM` type methods.

For your below suggestion:
> One quick-hack way out of the problem in this case would be to add a constant EJ to the Hamiltonian, which should enforce we have no negative eigenvalues.

That sounds interesting to try and, in principle, physically equivalent. I can try changing the source code of the Qubit class, for example.

The other hacky alternative I had was to allow all eigenvalues instead (using `evals_count`) and also eigenvectors, so that if we sort the output of `_SM`, we should get the same values as the default, I think.

Reply:
thanks.. yes, that's a good move "for now"...  adding the offset EJ to the main code, might confuse people, so we'd probably have to do it carefully... given you set up the core code with this PR, it will be easy to play and modify later what tests we re doing and how. 
i will take closer look at your code in a day or two.. and just merge once all ok.

Reply:
Great, thanks! So I'll refrain from changing anything in the main code here. I've added a few comments inside the tests to clarify why we are performing some "hacks" until the necessary bugs are fixed, which might be helpful later.